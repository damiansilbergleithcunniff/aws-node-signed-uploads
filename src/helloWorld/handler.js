import helloWorld from './helloWorld';

const main = (event, context, callback) => {
  const response = helloWorld(200, 'You should rework this component', event);

  callback(null, response);
};

export default main;
