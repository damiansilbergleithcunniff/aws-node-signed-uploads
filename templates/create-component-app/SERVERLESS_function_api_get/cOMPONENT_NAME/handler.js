import cOMPONENT_NAME from './cOMPONENT_NAME';

const main = (event, context, callback) => {
  const response = cOMPONENT_NAME(200, 'You should rework this component', event);

  callback(null, response);
};

export default main;
