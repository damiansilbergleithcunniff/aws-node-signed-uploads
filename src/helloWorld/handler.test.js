import handler from './handler';

describe('helloWorld handler', () => {
  test('generates expected response using SOMEVAL environment variable', () => {
    const tmp = process.env.SOMEVAL;
    process.env.SOMEVAL = 'SOME VALUE FROM ENV';

    let response;
    const event = 'EVENT';
    const callback = (_, res) => {
      response = res;
    };

    handler(event, null, callback);

    expect(response).toMatchSnapshot();

    // restore whatever was there
    process.env.SOMEVAL = tmp;
  });
});
