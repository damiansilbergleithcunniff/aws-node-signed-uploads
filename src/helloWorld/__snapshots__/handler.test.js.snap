// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helloWorld handler generates expected response 1`] = `
Object {
  "body": "{\\"message\\":\\"You should rework this component\\",\\"input\\":\\"EVENT\\"}",
  "statusCode": 200,
}
`;
